### ISSUE TITLE
Allow attaching an image to a top-level product

### ISSUE THREAD

**Issue by zenweasel** [reactioncommerce/reaction#3405]
The current behavior only allows you to attach an image to a product and a variant. But sometimes customers want to have a canonical image for a top-level product where all the variant images are specific to those variants. For example, maybe the top level image is an collage showing all the colors of a shoe, while the variant images are just of the single color.

This would also make it easier/simple for the grid to determine which image should be shown for a product.

I know the current behavior (not allowing this) was intentional but I am suggesting we change it.

These could just be represented by leaving the `variantId` blank in the `metadata`

**Comment by aaronjudd** [comment]
Actually, this is the original intended behavior, so 👍

**Comment by spencern** [comment]
Agree. This would make identifying images for the grid easier as well. I think Media Gallery 2.0 also needs to include some way to identify which images should be sent to the grid for different layouts and identify them with metadata as well - such as `"metadata.gridImage": "primary"`, `"metadata.gridImage": "secondary"`, `"metadata.gridImage: "none"` or something. Right now it's too difficult to filter images from the media gallery and we are still overpublishing to the client.
